source/docs/software/basic-programming/code-formatting.rst

```diff
@@ -0,0 +1,219 @@
+Using a Code Formatter
+======================
+
+Code formatters exist to ensure that the style of code written is consistent throughout the entire codebase. This is used in many major projects; from Android to OpenCV. Teams may wish to add a formatter throughout their robot code to ensure that the codebase maintains readability and consistency throughout.
+
+For this article, we will highlight using `Spotless <https://github.com/diffplug/spotless>`__ for Java teams and `wpiformat <https://github.com/wpilibsuite/styleguide/blob/main/wpiformat/README.rst>`__ for C++ teams.
+
+Spotless
+--------
+
+Configuration
+^^^^^^^^^^^^^
+
+Necessary ``build.gradle`` changes are required to get Spotless functional. In the ``plugins {}`` block of your ``build.gradle``, add the Spotless plugin so it appears similar to the below.
+
+.. code-block:: groovy
+
+   plugins {
+      id "java"
+      id "edu.wpi.first.GradleRIO" version "2021.2.1"
+      id 'com.diffplug.spotless' version '5.5.0'
+   }
+
+Spotless uses ``mavenCentral`` for distribution, so this needs to get added to a ``repositories {}`` block of your ``build.gradle``.
+
+.. note:: Please ensure ``repositories`` is not nested in any other block in your ``build.gradle``.
+
+.. code-block:: groovy
+
+   repositories {
+      mavenCentral()
+   }
```

Comment:
No, it doesn't. The `repostiories {}` block is for dependency repositories. Plugin repositories are defined in the `pluginManagement.repositories{}` block in `settings.gradle`. The spotless plugin is on the gradle plugin portal, which is in the template.